http/src/main/java/com/redhat/lightblue/client/http/transport/JavaNetHttpClient.java

```diff
@@ -0,0 +1,169 @@
+     */
+    public JavaNetHttpClient(ConnectionFactory connectionFactory, SSLSocketFactory sslSocketFactory) {
+        this.connectionFactory = Objects.requireNonNull(connectionFactory, "connectionFactory");
+        this.sslSocketFactory = sslSocketFactory;
+    }
+
+    public static JavaNetHttpClient fromLightblueClientConfiguration(LightblueClientConfiguration config)
+            throws GeneralSecurityException, IOException {
+        Objects.requireNonNull(config, "config");
+
+        SSLSocketFactory sslSocketFactory = config.useCertAuth()
+                ? SslSocketFactories.javaNetSslSocketFactory(config)
+                : null;
+
+        return new JavaNetHttpClient(sslSocketFactory);
+    }
+
+    @Override
+    public String executeRequest(LightblueRequest request, String baseUri) throws IOException {
+        String url = request.getRestURI(baseUri);
+        HttpURLConnection connection = connectionFactory.openConnection(url);
+
+        if (connection instanceof HttpsURLConnection) {
+            HttpsURLConnection httpsUrlConnection = (HttpsURLConnection) connection;
+
+            if (sslSocketFactory != null) {
+                httpsUrlConnection.setSSLSocketFactory(sslSocketFactory);
+            }
+        }
+
+        connection.setRequestMethod(request.getHttpMethod().toString());
+        connection.setRequestProperty("Accept", "application/json");
+        connection.setRequestProperty("Accept-Charset", "utf-8");
+
+        String body = request.getBody();
+        if (StringUtils.isNotBlank(body)) {
+            sendRequestBody(body, connection);
+        }
+
+        return response(connection);
+    }
+
+    @Override
+    public void close() throws IOException {
+        // Nothing to do
+    }
+
+    private void sendRequestBody(String body, HttpURLConnection connection)
+            throws IOException {
+        byte[] bodyUtf8Bytes = body.getBytes(UTF_8);
+        int length = bodyUtf8Bytes.length;
+
+        connection.setDoOutput(true);
+        connection.setFixedLengthStreamingMode(length);
+
+        connection.setRequestProperty("Content-Length", Integer.toString(length));
+        connection.setRequestProperty("Content-Type", "application/json; charset=utf-8");
+
+        try (OutputStream requestStream = connection.getOutputStream()) {
+            requestStream.write(bodyUtf8Bytes);
+        }
+    }
+
+    /**
+     * Parses response, whether or not the request was successful, if possible. Reads entire input
+     * stream and closes in order to so the socket knows it is finished and may be put back into a
+     * pool for reuse.
+     */
+    private String response(HttpURLConnection connection) throws IOException {
+        try (InputStream responseStream = connection.getInputStream()) {
+            return readResponseStream(responseStream, connection);
+        } catch (IOException e) {
+            try (InputStream errorResponseStream = connection.getErrorStream()) {
```

Comment:
Should that be responsibility of application? Or perhaps the `LightblueHttpClient` can examine response object for error codes and throw appropriate exceptions with response body and request info. I think #66 accomplished this (though I feel it's weird that an exception is only thrown when trying a client tries to use the response body, and not before the response is returned from the client).